text/nodejs/0014-yarn-berry.md

```diff
@@ -0,0 +1,280 @@
+# Support for Yarn "Berry" 
+
+## Summary
+
+The Paketo Node.js Buildpack supports building apps which use
+[Yarn](https://yarnpkg.com/) as a package/project management tool. Over the
+last few years, the developers of Yarn have released new major versions (v2 &
+v3), collectively called Yarn "Berry", which represent a completely new
+direction for managing dependencies in Node.js projects. Chief among the
+changes is the introduction of [Plug 'n
+Play](https://yarnpkg.com/features/pnp), a system which removes the reliance on
+`node_modules` and can eliminate the need to run `yarn install` at all. The
+Yarn implementation buildpacks do not currently support building apps which use
+Berry.  
+
+There have been quite a number of changes in the behaviour of Yarn with the
+advent of Berry. Along with performance improvements and optimizations, many
+CLI operations and methods of configuration have been deprecated or removed,
+including those crucial to the functioning of the buildpack as it exists today. 
+
+## Proposal
+
+Yarn Berry seems to represent the direction in which the project is headed, as
+[stated by its lead
+maintainer](https://dev.to/arcanis/introducing-yarn-2-4eh1#conclusion). Given
+this, and the extent of the changes to the CLI, this RFC proposes that a `yarn-berry-install`
+buildpack be created to enable the new installation workflows.
+
+## Motivation
+
+As developers continue to adopt Yarn Berry, demand for the Node.js buildpack to
+support it has steadily increased.
+
+
+## Implementation
+
+
+### API
+
+`yarn-berry-install`
+
+Provides: `yarn_cache` OR `node_modules` Requires: `node` during `build`
+
+The buildpack does not require `yarn` since the Berry release itself is meant
+to be checked into version control under `.yarn/releases`. Node v16.10 and
+later include
+[Corepack](https://nodejs.org/dist/latest/docs/api/corepack.html), which is the
+preferred method of managing package manager installations across projects. 
+
+
+A new order group will be added to the Nodejs language family buildpack,
+resulting in the following:
+
+```toml
+...
+
+[[order]]
+
+  [[order.group]]
+    id = "paketo-buildpacks/ca-certificates"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/watchexec"
+    optional = true
+
+  [[order.group]]
+    id = "paketo-buildpacks/node-engine"
+
+  [[order.group]]
+    id = "paketo-buildpacks/yarn-berry-install"
```

Comment:
Should this be optional and opt out if users are going with the [zero-installs](https://yarnpkg.com/features/zero-installs) method? This might mess up detection for the group without some changes in (yarn start), but it could be confusing to see yarn install opt-in when you are explicitly striving for zero-installs?

Reply:
This order group may now be obsolete as I was able to prove out a sane way to keep the build logic for both Yarn classic and Berry in a single buildpack (per @ryanmoran's suggestion). As for the UX, I figured we could indicate through the logs whenever the buildpack has chosen to forgo running `yarn install` based on adherence to the zero-installs setup.